.github/workflows/snapshot.yaml

```diff
@@ -0,0 +1,112 @@
+name: Publish snapshot
+
+on:
+  pull_request:
+    types: [labeled]
+
+env:
+  JAVA_VERSION: '17'
+
+jobs:
+
+  check-version:
+    if: ${{ contains(github.event.pull_request.labels.*.name, 'publish-snapshot') }}
+    runs-on: ubuntu-latest
+    outputs:
+      snapshot-version: ${{ steps.version-step.outputs.version }}
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Setup Java
+        uses: actions/setup-java@v4
+        with:
+          distribution: 'temurin'
+          java-version: ${{ env.JAVA_VERSION }}
+
+      - name: Setup Gradle
+        uses: gradle/gradle-build-action@v3
+
+      - name: Get DDI-Lifecycle lib version
+        id: version-step
+        run: |
+          ./gradlew # (load the gradle wrapper, required for the get version step)
+          echo "version=$(./gradlew :model:printVersion --console=plain -q)" >> $GITHUB_OUTPUT
+
+      - name: Snapshot version verification
+        run: |
+          if ! [[ "${{ steps.version-step.outputs.version }}" =~ ^[0-9]+.[0-9]+.[0-9]+-SNAPSHOT$ ]]; then
```

Comment:
I quite like enforcing "snapshot" since it has a meaning for maven (snapshots have a different overwrite mechanism), and adding other tags introduces some complexity that often feels not necessary 

Yet it is restrictive in regards to semver... so we could use the more complex regex.

Isn't the "+" symbol missing for metadata?

Reply:
I propose this pattern : `(?>^[0-9]+.[0-9]+.[0-9]+-SNAPSHOT$)|(?>^[0-9]+.[0-9]+.[0-9]+-SNAPSHOT\+[0-9A-Za-z-]+$)`

it means that the version must match `^[0-9]+.[0-9]+.[0-9]+-SNAPSHOT$` (for example 1.1.1-SNAPSHOT) or must match `^[0-9]+.[0-9]+.[0-9]+-SNAPSHOT\+[0-9A-Za-z-]+$` (for example 1.1.1-SNAPSHOT+feat1). So it is mandatory the version starts with x.y.z-SNAPSHOT and then it may be suffixed by a string wich starts with +

Reply:
great!